src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,29 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (try
+          (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
@camlow325 this implementation actually does perform all stops, even if one errors. `run-lifecycle-fns` calls a doseq and catches exceptions, logging them as errors as it goes (almost identially to the body of `shutdown!`. I can push out an updated test still, because i don't think my test captured this well.